### PR TITLE
Remove `configOverrides` option (#5522)

### LINKED ISSUE #5522: Remove `configOverrides` option
[`configOverrides`](https://stylelint.io/user-guide/usage/node-api/#configoverrides) in Node.js API would be quite confusing with [`overrides`](https://github.com/stylelint/stylelint/issues/3128) in config.

I propose to remove it in v14.

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

Closes #5522

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const EXIT_CODE_ERROR = 2;
  * @property {boolean} [reportInvalidScopeDisables]
  * @property {boolean} [reportDescriptionlessDisables]
  * @property {number} [maxWarnings]
- * @property {string | boolean} quiet
+ * @property {boolean} quiet
```
Comment (hudochenkov) [inline lib/cli.js:43]: Don't know why it was `string`. Seems to me no different to other booleans.

**Review by jeddy3 — APPROVED** [review]
LGTM, thanks!

(Failing test is an unrelated flaky one.)

File: lib/augmentConfig.js [hunk 2]
```diff
@@ -3,7 +3,6 @@
 const configurationError = require('./utils/configurationError');
 const getModulePath = require('./utils/getModulePath');
 const globjoin = require('globjoin');
-const merge = require('deepmerge');
```
Comment (ybiquitous) [inline lib/augmentConfig.js:6]: [suggestion] Oh, it seems we now use `deepmerge` only for testing. Can we move the package to `devDependencies`?

```console
$ git grep deepmerge
lib/__tests__/disableRanges.test.js:4:const merge = require('deepmerge');
lib/testUtils/mergeTestDescriptions.js:3:const merge = require('deepmerge');
package-lock.json:16:        "deepmerge": "^4.2.2",
package-lock.json:2839:    "node_modules/deepmerge": {
package-lock.json:2841:      "resolved": "https://registry.npmjs.org/deepmerge/-/deepmerge-4.2.2.tgz",
package-lock.json:5385:        "deepmerge": "^4.2.2",
package-lock.json:14726:    "deepmerge": {
package-lock.json:14728:      "resolved": "https://registry.npmjs.org/deepmerge/-/deepmerge-4.2.2.tgz",
package-lock.json:16651:        "deepmerge": "^4.2.2",
package.json:114:    "deepmerge": "^4.2.2",
```

https://github.com/stylelint/stylelint/blob/5d7695af46f02bf380b47e358173aeb190c81e97/package.json#L114

Comment (hudochenkov) [reply lib/augmentConfig.js:6]: You always spot all these subtle details! Great idea! Fixed.

Comment (ybiquitous) [reply lib/augmentConfig.js:6]: Thank you. 9eef5ac looks good! 👍🏼

**Review by ybiquitous — APPROVED** [review]
LGTM! 👏🏼

**Comment by ybiquitous** [comment]
I've retried CI to fix the flaky test failure.

**Comment by jeddy3** [comment]
> - Removed: `configOverrides` option ([#5530](https://github.com/stylelint/stylelint/pull/5530)).

**Pull request merged**